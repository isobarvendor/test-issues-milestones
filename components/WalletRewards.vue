<template>
  <div class="wallet-rewards" >
    <SliderHeader>{{cmsData.Wallets.RewardsTitle}}</SliderHeader>
    <div class="wallet-desktop" v-if="$mq == 'xl'">
      <div class="wallet-swiper" v-for="(item, index) in toBeShown" :key="'reward'+index">
        <div class="wallet-swiper-item">
          <img src="/img/prize-default.svg"
          />
        </div>
        <div class="description bg">
            <div class="name">{{item.currency_name}}</div>
            <div class="expiry">
              <span>365</span>
              <span>days left</span>
            </div>
          </div>
      </div>
    </div>
    <swiper class="swiper wallet-swiper" :options="swiperOption"  v-if="$mq == 'sm' || $mq == 'md' || $mq == 'lg'">
      <swiper-slide
        v-for="(item, index) in data.expirationWallet"
        :key="'reward'+index"
        class="wallet-swiper-item"
      >
        <img
          :src="item.imageMobile ? item.imageMobile.url : '/img/prize-default.svg'"
        />
        <div class="description bg">
          <div class="name">{{item.currency_name}}</div>
          <div class="expiry">
            <span>{{dayDiff(dates)}}</span>
            <span>days left</span>
          </div>
        </div>
      </swiper-slide>
      <div class="prize-swiper-pagination" slot="pagination"></div>
    </swiper>
    <a v-if="$mq == 'xl'" class="button center" @click="nextPage" :disabled="currentPage == totalPages">View all</a>
  </div>
</template>

<script>
import SliderHeader from "@/components/WalletSliderHeader";
import { Swiper, SwiperSlide } from "vue-awesome-swiper";
export default {
  name: "WalletRewards",
  components: {
    SliderHeader,
    Swiper,
    SwiperSlide,
  },
  data() {
    return {
      swiperOption: {
        slidesPerView: 2,
        spaceBetween: 10,
        breakpoints: {
          // when window width is >= 320px

          768: {
            slidesPerView: 3,
            spaceBetween: 20,
          },
          992: {
            slidesPerView: 3,
            spaceBetween: 30,
          },
        },
        freeMode: true,
        pagination: {
          el: ".prize-swiper-pagination",
          clickable: true,
          bulletClass: "prize-swiper-pagination-bullet",
          bulletActiveClass: "prize-swiper-pagination-bullet-active",
        }
      },
      currentPage: 1
    };
  },
  props: {
    data: null,
    cmsData:null
  },
  computed:{
  	toBeShown() {
    	return this.data.expirationWallet.slice(0, this.currentPage * 3);
    },
    totalPages() {
    	return Math.ceil( this.data.expirationWallet.length / 3);
    }
  },
  methods:{
    nextPage(){
    	if(this.currentPage <  this.totalPages) this.currentPage++;
    },
    prevPage(){
    	this.currentPage = this.currentPage - 1 || 1;
    },
    dayDiff(dates){
      let today = new Date();
      let dates2 = new Date(dates);
      let dateDiff = today.getDate() - dates2.getDate();
      return dateDiff
    }
  },
  beforeMount() {},
  mounted(){

  }
};
</script>

<style lang="scss">
.wallet-rewards{
    margin-bottom: 40px;
    .wallet-desktop{
      display: flex;
      flex-wrap: wrap;
      .wallet-swiper{
        margin: 30px 20px;
        width: 29%;
      }
      .description{
        padding: 20px;
        border-radius: 0px 0px 20px 20px;

        .name:after{
          content: '';
          display: block;
          margin: .5em 0;
          height: 1px;
          background: #585858;

        }
        .expiry{
          display: flex;
          justify-content: flex-end;
          span{
            font-size:14px;
          }
          span:first-child{
            margin-right: 0.5em;
            font-weight: bold;
            font-size: 26px;
          }
        }
      }
    }
    .prize-swiper-pagination > * + *:before{
      background: #585858
    }
    .prize-swiper-pagination .prize-swiper-pagination-bullet{
      background: #585858
    }
    .prize-swiper-pagination .prize-swiper-pagination-bullet-active{
      background: #ffffff
    }
    @media only screen and (max-width: 1199px) {
      .swiper.wallet-swiper{
        .description{
          padding: 1em 10px;
          .name{
            font-size: 14px;
          }
          .expiry{
            justify-content: flex-start;
            span:first-child{
              font-size: 14px;
            }
          }
        }

      }
    }

}
</style>
