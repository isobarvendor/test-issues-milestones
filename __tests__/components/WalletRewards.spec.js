
import { mount } from '@vue/test-utils'

import WalletRewards from "@/components/WalletRewards";
import SliderHeader from "@/components/WalletSliderHeader";
import { Swiper, SwiperSlide } from "vue-awesome-swiper";

let $mq ="lg";
const factory = () => {
  return mount(WalletRewards, {
    mocks: {
      $mq:$mq
    },
    data() {
      return {
        swiperOption: {
          slidesPerView: 2,
          spaceBetween: 10,
          breakpoints: {
            // when window width is >= 320px

            768: {
              slidesPerView: 3,
              spaceBetween: 20,
            },
            992: {
              slidesPerView: 3,
              spaceBetween: 30,
            },
          },
          freeMode: true,
          pagination: {
            el: ".prize-swiper-pagination",
            clickable: true,
            bulletClass: "prize-swiper-pagination-bullet",
            bulletActiveClass: "prize-swiper-pagination-bullet-active",
          }
        },
        currentPage: 1
      };
    },
    propsData: {
      data :{
        "expirationWallet": [
          {
              "gpp_user_id": "UUID|CDS region",
              "amount": 20,
              "currency_id": "coin",
              "currency_name": "CCA-coin",

              "configuration_id": "exampleConfig123",
              "last_modified": 1557764018189,
              "valid_thru": 1355228281846,
              "already_spent": 15,
              "spent_amount": 0
          },
          {
              "gpp_user_id": "UUID|CDS region",
              "amount": 1,
              "configuration_id": "exampleConfig123",
              "currency_id": "coin",
              "currency_name": "CCA-coin",
              "valid_thru": 1581811199000,
              "already_spent": 0,
              "spent_amount": 0
          },
          {
              "gpp_user_id": "UUID|CDS region",
              "amount": 40,
              "currency_id": "coin",
              "currency_name": "CCA-coin",

              "configuration_id": "exampleConfig123",
              "last_modified": 1557764018189,
              "valid_thru": 1345228281846,
              "already_spent": 0,
              "spent_amount": 0
          }
      ]

      },
      cmsData:{
        Wallets:{
          RewardsTitle :"test"
        }
      }
    }
  });
};

describe("WalletRewards", () => {


  test('renders a SliderHeader', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(SliderHeader);
    expect(component.exists()).toBeTruthy();
  });

  test('renders a Swiper', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(Swiper);
    expect(component.exists()).toBeTruthy();
  });

  test('renders a SwiperSlide ', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(SwiperSlide);
    expect(component.exists()).toBeTruthy();
  });



  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
