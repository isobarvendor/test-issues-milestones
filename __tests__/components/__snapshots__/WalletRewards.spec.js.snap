// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletRewards renders properly 1`] = `
<div class="wallet-rewards">
  <div class="slider-header">
    <div class="title">test</div>
  </div>
  <!---->
  <div class="swiper-container swiper wallet-swiper swiper-container-initialized swiper-container-horizontal swiper-container-free-mode">
    <div class="swiper-wrapper" style="webkit-transition-duration: 0ms; transition-duration: 0ms; webkit-transform: translate3d(NaNpx, 0px, 0px); transform: translate3d(NaNpx, 0px, 0px);">
      <div class="wallet-swiper-item swiper-slide"><img src="/img/prize-default.svg">
        <div class="description bg">
          <div class="name">CCA-coin</div>
          <div class="expiry"><span>NaN</span> <span>days left</span></div>
        </div>
      </div>
      <div class="wallet-swiper-item swiper-slide"><img src="/img/prize-default.svg">
        <div class="description bg">
          <div class="name">CCA-coin</div>
          <div class="expiry"><span>NaN</span> <span>days left</span></div>
        </div>
      </div>
      <div class="wallet-swiper-item swiper-slide"><img src="/img/prize-default.svg">
        <div class="description bg">
          <div class="name">CCA-coin</div>
          <div class="expiry"><span>NaN</span> <span>days left</span></div>
        </div>
      </div>
    </div>
    <div class="prize-swiper-pagination"></div><span class="swiper-notification" aria-live="assertive" aria-atomic="true"></span>
  </div>
  <!---->
</div>
`;
